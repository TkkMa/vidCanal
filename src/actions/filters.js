import database from '../firebase/firebase';

export const setSortBy = (text='') => ({
    type: 'SET_SORT_BY',
    text
});

export const setUploadDate = (text='') => ({
    type: 'SET_UPLOAD_DATE',
    text
});

export const setResPerPage = (num)=>({
    type: 'SET_RESULTS_PER_PAGE',
    num
})

export const setFavCount = (favCount)=>({
    type: 'SET_FAV_COUNT',
    favCount
})

export const addFavCount = (countData={})=>{
    return(dispatch,getState) =>{
        const uid = getState().auth.uid;
        const {
            count = 0,
            videoId = '', 
            ids =[],
        } = countData;
        return database.ref(`users/${uid}/unSeenLikes`).push({videoId}).then((ref)=>{
            dispatch(setFavCount({
                count,
                ids: ids.concat({DB_id: ref.key, videoId})
            }));
        });
    };
};

export const removeFavCount = (countData={})=>{
    return(dispatch, getState) =>{
        const uid = getState().auth.uid;
        const {
            count=0,
            videoIds=[],
            foundDbId=''
        } = countData;
        return database.ref(`users/${uid}/unSeenLikes/${foundDbId}`).remove().then(()=>{
            dispatch(setFavCount({
                count,
                ids: videoIds 
            }));
        });
    };
};

export const clearFavCount = () =>{
    return(dispatch, getState) =>{
        const uid = getState().auth.uid;
        return database.ref(`users/${uid}/unSeenLikes`).remove().then(()=>{
            dispatch(setFavCount({
                count:0,
                ids: []
            }));
        });
    };
};

export const loadFavCount =()=>{
    return(dispatch, getState)=>{
        const uid = getState().auth.uid;
        return database.ref(`users/${uid}/unSeenLikes`)
                .once('value')
                .then((snapshot)=>{
                    const videoIds = [];

                    snapshot.forEach((childSnapshot)=>{
                        videoIds.push({
                            DB_id: childSnapshot.key,
                            ...childSnapshot.val()
                        });
                    });
                    if(videoIds.length){
                        dispatch(setFavCount({
                            count: videoIds.length,
                            ids: videoIds
                        }));
                    }
                })

    }
};

// SET_START_DATE
export const setStartDate = (startDate) => ({
    type: 'SET_START_DATE',
    startDate
  });
  
// SET_END_DATE
export const setEndDate = (endDate) => ({
    type: 'SET_END_DATE',
    endDate
});

// SET_TEXT_FILTER
export const setTextFilter = (text = '') => ({
    type: 'SET_TEXT_FILTER',
    text
});

// TOGGLE_ISSAVED
export const toggleIsSavedFilter = (isSaved)=>({
    type: 'TOGGLE_ISSAVED_FILTER',
    isSaved
})

// TOGGLE_ISSEARCHBYKEY
export const toggleIsSavedFilter = (isSearchByKey)=>({
    type: 'TOGGLE_ISSEARCH_BY_KEY_FILTER',
    isSearchByKey
})