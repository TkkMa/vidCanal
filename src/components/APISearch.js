const axios = require('axios');
const moment = require('moment');

const ROOT_URL = {
    YT: 'https://www.googleapis.com/youtube/v3/search',
    D: 'https://api.dailymotion.com/videos',
    V: 'https://api.vimeo.com/videos'
}
const ROOT_URL_VIDEO = {
    YT: 'https://www.googleapis.com/youtube/v3/videos'
    // D: 'https://api.dailymotion.com/video/',
    // V: 'https://api.vimeo.com/videos'
};
const API_KEY= {
    YT:'AIzaSyDWU_q-QSxsSUECDdeWI8WBnVE2q07p804',
    V: 'f0ca0fbcd4918f82208f732dfded4341'
}

const vidAPISearch = (options={}, engine='YT')=>{
    let params;
    if (engine === 'YT'){
        params = {
            maxResults: options.num,
            part: 'snippet',
            key: API_KEY.YT,
            q: options.term,
            type: 'video',
            order: options.sortByVal,
            pageToken: options.pageToken[engine]
        };
        if(options.sortTimeVal){params.publishedAfter=options.sortTimeVal;}
        return axios.get(ROOT_URL[engine], { params: params })
            .then(response=>{           
                return {
                    pageInfo: response.data.pageInfo, 
                    items: response.data.items, 
                    nextPageToken: response.data.nextPageToken || ''
                };
            })
            .catch(function(error) {
                console.error(error);
                return null;
            });
    } else if(engine==='D'){
        params = {
            limit: options.num,
            search: options.term,
            sort: options.sortByVal,
            page: options.pageToken[engine],
            fields: 'id,title,owner.screenname,owner.url,views_total,embed_url,created_time,description,thumbnail_60_url,thumbnail_120_url,thumbnail_180_url'
          };

        if(options.sortTimeVal){params.created_after=moment(options.sortTimeVal).unix();}
        if(!params.sort){console.error('No alphabetical sorting exists for Dailymotion'); return null}
        return axios.get(ROOT_URL[engine], {params: params})
            .then(response=>{
                const arrayOfObj = response.data.list.map(({
                    ['owner.screenname']: ownerScreenname,
                    ['owner.url']: ownerUrl,
                    ...rest
                })=>({
                    ownerScreenname,
                    ownerUrl,
                    ...rest
                }))
                return {
                    pageInfo: {
                        total: response.data.total,
                        has_more: response.data.has_more
                    }, 
                    items: arrayOfObj, 
                    nextPageToken: options.pageToken[engine]+1
                };            
            })
            .catch(function(error) {
                console.error(error);
                return null;
            });
    } else if(engine==='V'){
        params = {
            access_token: API_KEY.V,
            per_page: options.num,
            query: options.term,
            page: options.pageToken[engine],
            sort: options.sortByVal,
            fields:'total,page,per_page,paging,data,uri,pictures,name,release_time,user,stats,description,embed'
        }

        // if(options.sortTimeVal){params.filter=moment(options.sortTimeVal).unix();}
        return axios.get(ROOT_URL[engine], { params: params })
            .then(response=>{
                return {
                    pageInfo: {
                        paging: response.data.paging,
                        total: response.data.total
                    }, 
                    items: response.data.data, 
                    nextPageToken: options.pageToken[engine]+1
                };
            })
            .catch(function(error) {
                console.error(error);
                return null;
            });
    }  
}

//-- Daily motion and Vimeo only requires one API Call in the previous search request to get all fields
//-- Youtube requires two API calls
const indVidAPISearch = (options, engine='YT') =>{
    let params;
    if(engine==='YT'){
        params={
            id: options.id,
            key: API_KEY.YT,
            part: 'snippet, statistics'
        }

        return axios.get(ROOT_URL_VIDEO[engine], {params: params})
            .then(response=>{
            return response.data.items
            })
            .catch(function(error) {
            console.error(error);
        });
    } else if(engine==='D'){
        return [options];
    } else if(engine==='V'){
        return [options];
    }
};

module.exports = {
    vidAPISearch,
    indVidAPISearch
}