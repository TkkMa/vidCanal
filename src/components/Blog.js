import React from 'react';
import { Link } from 'react-router-dom';

const Blog = () => (
  <div className="construction">
    <img src='/images/under-construction-image.png' />
    <Link to="/">Go home</Link>
  </div>
);

export default Blog;