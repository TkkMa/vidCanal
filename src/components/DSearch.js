const axios = require('axios');
const ROOT_URL = 'https://api.dailymotion.com/videos';
// const ROOT_URL_VIDEO = 'https://www.googleapis.com/youtube/v3/videos';
// const API_KEY='AIzaSyDWU_q-QSxsSUECDdeWI8WBnVE2q07p804';

const DSearch = (options={})=>{

  const params = {
    limit: options.num,
    search: options.term,
    sort: options.sortBy,
    page: options.pageToken
  };
  if(options.sortTimeVal){params.createdAfter=options.sortTimeVal;}

  return axios.get(ROOT_URL, { params: params })
    .then((response)=>{
      return {
        has_more: response.data.has_more,
        items: response.data.list, 
        nextPageToken: response.data.page + 1
      };
    })
    .catch(function(error) {
      console.error(error);
    });  
}

const YTVideo = (options) =>{
  const params={
    id: options.id,
    key: API_KEY,
    part: 'snippet, statistics'
  }

  return axios.get(ROOT_URL_VIDEO, {params: params})
    .then((response)=>{
      return response.data.items
    })
    .catch(function(error) {
      console.error(error);
    });
}

module.exports = {
  YTSearch,
  YTVideo
}