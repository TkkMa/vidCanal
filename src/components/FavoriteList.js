import React, {Component} from 'react';
import {connect} from 'react-redux';
import FavoriteListItem from './FavoriteListItem';
import FavoriteModal from './FavoriteModal';
import favorites from '../selectors/favorites';
// import {setFavCount} from '../actions/filters';

export class FavoriteList extends Component {

    state ={
        showModal: false,
        clickedVideo: {}
    }

    handleOpenModal= (clickedVideo)=>{
        this.setState({
            showModal: true,
            clickedVideo
        });
    }

    handleCloseModal=()=>{
        this.setState({showModal: false});
    }

    render(){
        const filteredVids = this.props.uniqueLikedVideos;
        return(
                <div className="row">
                    {
                        (filteredVids.length === 0) ? (
                            <p>No favorite videos</p>
                        ):(
                            <div className="col s12">
                                <ul className="collection">
                                    {
                                        filteredVids.map((video, index) =>(
                                            <FavoriteListItem 
                                                {...this.props} 
                                                handleOpenModal={this.handleOpenModal}
                                                key={video.id+index} 
                                                video={video}
                                            />
                                        ))
                                    }
                                </ul>
                            </div>
                        )
                    }
                    <FavoriteModal
                        showModal={this.state.showModal}
                        handleCloseModal={this.handleCloseModal}
                        video={this.state.clickedVideo}
                    />
                </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
      visitedVideos: state.videos.visitedVideos,
      uniqueLikedVideos: favorites(state.videos.visitedVideos)
    };
};

export default connect(mapStateToProps)(FavoriteList);