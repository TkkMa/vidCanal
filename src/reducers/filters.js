import moment from 'moment';

// startDate & endDate for DatePicker
const filtersReducerDefaultState = {
    sortBy: 'relevance',
    uploadDate: 'allTime',
    startDate: moment().startOf('month'),
    endDate: moment(),
    text: '',
    resultsPerPage: 10,
    isSaved: false,
    isSearchByKey: false,
    unViewedFavCount: 0,
    unViewedFavIds:[]
};

export default (state=filtersReducerDefaultState, action) =>{
    switch(action.type){
        case 'SET_SORT_BY':
            return{
                ...state,
                sortBy: action.text
            };
        case 'SET_UPLOAD_DATE':
            return{
                ...state,
                uploadDate: action.text
            };
        case 'SET_RESULTS_PER_PAGE':
            return{
                ...state,
                resultsPerPage: action.num
            };
        case 'SET_FAV_COUNT':
            return{
                ...state,
                unViewedFavCount: action.favCount.count,
                unViewedFavIds: action.favCount.ids
            };
        case 'SET_START_DATE':
            return {
                ...state,
                startDate: action.startDate
            };
        case 'SET_END_DATE':
            return {
                ...state,
                endDate: action.endDate
            };
        case 'SET_TEXT_FILTER':
            return{
                ...state,
                text: action.text
            }
        case 'TOGGLE_ISSAVED_FILTER':
            return{
                ...state,
                isSaved: action.isSaved
            }
        case 'TOGGLE_ISSEARCH_BY_KEY_FILTER':
            return{
                ...state,
                isSearchByKey: action.isSearchByKey
            }
        default:
            return state;
    };
}