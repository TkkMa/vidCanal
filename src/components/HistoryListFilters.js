import _ from 'lodash';
import React, {Component} from 'react'
import moment from 'moment';
import {connect} from 'react-redux';
import 'react-dates/initialize';
import {DateRangePicker} from 'react-dates';
import {setStartDate, setEndDate, setTextFilter, toggleIsSavedFilter, toggleSearchKeyFilter} from '../actions/filters';
import DeletionModal from './DeletionModal';
import {startClearVideoHistory} from '../actions/videos';

export class HistoryListFilters extends Component{
    
    state={
        calendarFocused: null,
        showModal: false,
        message: 'video history'
    }

    componentDidMount(){
        $('.sidenav').sidenav({
            preventScrolling: false
        })
    }

    onDatesChange = ({startDate, endDate}) =>{
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }

    onFocusChange = (calendarFocused)=>{
        this.setState(()=>({calendarFocused}));
    }

    onTextChange = (e)=>{
        this.props.setTextFilter(e.target.value);
    }

    onSwitchChange = ()=>{
        this.props.toggleIsSavedFilter(!this.props.filters.isSaved);
    }

    submitHandler= (e)=>{
        e.preventDefault();
    }
    
    clearText= (e)=>{
        $(e.target).siblings('#search').val('');
        this.props.setTextFilter('');
    }

    handleOpenModal=()=>{
        this.setState({showModal: true});
    }

    handleCloseModal=()=>{
        this.setState({showModal: false});
    }

    clearList = ()=>{
        this.props.startClearVideoHistory();
        this.handleCloseModal();
    }

    render(){
        const onTextChange = _.debounce((text)=>{this.onTextChange(text)}, 300);
        return(
            <div>
                <nav className='HLF-1'>
                    <div className="nav-wrapper row">
                        <div className="col s12 m8 l6 xl5">
                            <DateRangePicker
                                startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
                                startDateId="your_unique_start_date_id" // PropTypes.string.isRequired,
                                endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
                                endDateId="your_unique_end_date_id" // PropTypes.string.isRequired,
                                onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
                                focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                                onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
                                showClearDates={true}
                                isOutsideRange={day=> moment(day).startOf('day').isAfter()}
                                withPortal={true}
                                displayFormat={()=> "DD/MM/YYYY"}
                            />
                        </div>
                        <div className="col s4 m4 l2 xl2">
                            <div className="switch">
                                <label>
                                <input 
                                    onChange={this.onSwitchChange} 
                                    defaultChecked={this.props.filters.isSaved}
                                    type="checkbox" 
                                />
                                <span className="lever"></span>
                                    Liked
                                </label>
                            </div>
                        </div>
                        <div className="col s6 m10 l2 xl3">
                            <form id="historyForm" onSubmit={this.submitHandler} className="right">
                                <div id="historySearchBar" className="right input-field">
                                    <input 
                                        onChange={this.onTextChange}
                                        value={this.props.filters.text} 
                                        id="search" type="search" required
                                        placeholder="Search term" 
                                    />
                                    <label className="label-icon" for="search"><i className="material-icons">search</i></label>
                                    <i className="material-icons" onClick={this.clearText}>close</i>
                                </div>
                            </form>
                        </div>
                        <div className="col s1 m1 l1 xl1">
                            <button data-target="slide-out" className="sidenav-trigger btn btn-flat">
                                <i className="material-icons">list</i>
                                {(this.props.filters.searchKeyFilter.length)? <i className="material-icons">check_circle</i>: <i/>}
                            </button>
                        </div>
                        <div className="col s1 m1 l1 xl1">
                            <button onClick={this.handleOpenModal} className="btn btn-flat">
                                <i className="material-icons">delete_forever</i>
                            </button>
                        </div>
                    </div>
                </nav>
                <DeletionModal
                    showModal={this.state.showModal}
                    handleCloseModal={this.handleCloseModal}
                    clearList={this.clearList}
                    message={this.state.message}
                />
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch)=>({
    setStartDate: (startDate)=> dispatch(setStartDate(startDate)),
    setEndDate: (endDate)=> dispatch(setEndDate(endDate)),
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    toggleIsSavedFilter: (isSaved)=> dispatch(toggleIsSavedFilter(isSaved)),
    toggleSearchKeyFilter: (isSearchByKey)=> dispatch(toggleSearchKeyFilter(isSearchByKey)),
    startClearVideoHistory: ()=> dispatch(startClearVideoHistory())
})

export default connect(mapStateToProps, mapDispatchToProps)(HistoryListFilters);
