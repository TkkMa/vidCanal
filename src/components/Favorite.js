import React from 'react';
import FavoriteListFilters from './FavoriteListFilters';
import FavoriteList from './FavoriteList';

const Favorite = ()=>{
    return(
        <div className="container">
            <FavoriteListFilters />
            <FavoriteList />
        </div>
    )
}

export default Favorite