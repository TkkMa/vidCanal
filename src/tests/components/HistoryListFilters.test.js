import React from 'react';
import {shallow} from 'enzyme';
import {HistoryListFilters} from '../../components/HistoryListFilters';
import filters from '../fixtures/filters';
import moment from 'moment';

test('should render history list filter bar correctly', ()=>{

    const wrapper = shallow(<HistoryListFilters 
                                filters={filters} 
                            />);
    expect(wrapper).toMatchSnapshot();
});

test('should filter by likes', ()=>{
    const onSwitchChangeSpy = jest.fn();
    const wrapper = shallow(<HistoryListFilters 
                                filters={filters} 
                                toggleIsSavedFilter={onSwitchChangeSpy}    
                            />);
    const initialCheckedState = wrapper.state('checked');
    wrapper.find('input').at(0).simulate('change');
    expect(onSwitchChangeSpy).toHaveBeenLastCalledWith(!initialCheckedState);
    expect(wrapper.state('checked')).toBe(!initialCheckedState);                        
});

test('should set filtered text on input change', ()=>{
    const value = 'redux';
    const onTextChangeSpy = jest.fn();
    const wrapper = shallow(<HistoryListFilters 
                                filters={filters}
                                setTextFilter={onTextChangeSpy}
                            />);
    wrapper.find('input').at(1).simulate('change',{
        target : {value}
    });
    expect(onTextChangeSpy).toHaveBeenLastCalledWith(value);
});

test('should clear text on close icon click', ()=>{
    const value = '';
    const onTextChangeSpy = jest.fn();
    const wrapper = shallow(<HistoryListFilters 
                                filters={filters}
                                setTextFilter={onTextChangeSpy}    
                            />);
    wrapper.find('i').at(1).simulate('click', {target:"#historySearchBar i"});
    expect(wrapper.find('input#search').props().value).toBe('');
    expect(onTextChangeSpy).toHaveBeenLastCalledWith(value);
})

test('show modal upon clicking delete', ()=>{
    const wrapper = shallow(<HistoryListFilters filters={filters}/>);
    wrapper.find('i').at(2).simulate('click');
    expect(wrapper.state('showModal')).toBe(true);
})

test('should set new date on date change', ()=>{
    const setStartDateSpy = jest.fn();
    const setEndDateSpy = jest.fn();
    const now = moment();
    const early = moment();
    const wrapper = shallow(<HistoryListFilters
                                filters={filters}
                                setStartDate={setStartDateSpy}
                                setEndDate={setEndDateSpy}
                            />);
    wrapper.find('withStyles(DateRangePicker)').prop('onDatesChange')({startDate:early, endDate:now});
    expect(setStartDateSpy).toHaveBeenLastCalledWith(early);
    expect(setEndDateSpy).toHaveBeenLastCalledWith(now);
})

test('should set calendar focus on change', ()=>{
    const wrapper = shallow(<HistoryListFilters
        filters={filters}
    />);
    const calendarFocused = true;
    wrapper.find('withStyles(DateRangePicker)').prop('onFocusChange')(calendarFocused);
    expect(wrapper.state('calendarFocused')).toBe(calendarFocused);
})