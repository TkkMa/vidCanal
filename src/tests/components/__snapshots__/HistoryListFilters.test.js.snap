// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render history list filter bar correctly 1`] = `
<div>
  <nav
    className="HLF-1"
  >
    <div
      className="nav-wrapper row"
    >
      <div
        className="col s12 m8 l6 xl5"
      >
        <withStyles(DateRangePicker)
          anchorDirection="left"
          appendToBody={false}
          block={false}
          calendarInfoPosition="bottom"
          customArrowIcon={null}
          customCloseIcon={null}
          customInputIcon={null}
          daySize={39}
          disableScroll={false}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          endDate={"1970-01-01T00:00:00.000Z"}
          endDateId="your_unique_end_date_id"
          endDatePlaceholderText="End Date"
          firstDayOfWeek={null}
          focusedInput={null}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          initialVisibleMonth={null}
          inputIconPosition="before"
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepFocusOnInput={false}
          keepOpenOnDateSelect={false}
          minimumNights={1}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          noBorder={false}
          numberOfMonths={2}
          onClose={[Function]}
          onDatesChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          openDirection="down"
          orientation="horizontal"
          phrases={
            Object {
              "calendarLabel": "Calendar",
              "chooseAvailableEndDate": [Function],
              "chooseAvailableStartDate": [Function],
              "clearDates": "Clear Dates",
              "closeDatePicker": "Close",
              "dateIsSelected": [Function],
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "focusStartDate": "Interact with the calendar and add the check-in date for your trip.",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month.",
              "jumpToPrevMonth": "Move backward to switch to the previous month.",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
              "moveFocusByOneMonth": "Switch months.",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
              "openThisPanel": "Open this panel.",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field.",
              "selectFocusedDate": "Select the date in focus.",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          readOnly={false}
          regular={false}
          renderCalendarInfo={null}
          renderDayContents={null}
          renderMonthElement={null}
          renderMonthText={null}
          reopenPickerOnClearDates={false}
          required={false}
          screenReaderInputMessage=""
          showClearDates={true}
          showDefaultInputIcon={false}
          small={false}
          startDate={"1969-12-31T16:00:00.000Z"}
          startDateId="your_unique_start_date_id"
          startDatePlaceholderText="Start Date"
          verticalHeight={null}
          verticalSpacing={22}
          weekDayFormat="dd"
          withFullScreenPortal={false}
          withPortal={true}
        />
      </div>
      <div
        className="col s4 m4 l2 xl2"
      >
        <div
          className="switch"
        >
          <label>
            <input
              defaultChecked={false}
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="lever"
            />
            Liked
          </label>
        </div>
      </div>
      <div
        className="col s7 m11 l3 xl4"
      >
        <form
          className="right"
          id="historyForm"
          onSubmit={[Function]}
        >
          <div
            className="right input-field"
            id="historySearchBar"
          >
            <input
              id="search"
              onChange={[Function]}
              placeholder="Search term"
              required={true}
              type="search"
              value=""
            />
            <label
              className="label-icon"
              for="search"
            >
              <i
                className="material-icons"
              >
                search
              </i>
            </label>
            <i
              className="material-icons"
              onClick={[Function]}
            >
              close
            </i>
          </div>
        </form>
      </div>
      <div
        className="col s1 m1 l1 xl1"
      >
        <i
          className="material-icons"
          onClick={[Function]}
        >
          delete_forever
        </i>
      </div>
    </div>
  </nav>
  <DeletionModal
    clearList={[Function]}
    handleCloseModal={[Function]}
    message="video history"
    showModal={false}
  />
</div>
`;
